src/block/Campfire.php

```diff
@@ -0,0 +1,264 @@
+		$this->position->getWorld()->setBlock($this->position, $this->setExtinguished(false));
+		$this->position->getWorld()->scheduleDelayedBlockUpdate($this->position, 1);
+	}
+
+	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		if($this->getSide(Facing::DOWN)->isTransparent()){
+			return false;
+		}
+		return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
+	}
+
+	public function onBreak(Item $item, ?Player $player = null) : bool{
+		$this->items = [];
+		$this->itemTime = [];
+		return parent::onBreak($item, $player);
+	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		if($player !== null){
+			if($item instanceof FlintSteel){
+				if($this->extinguished){
+					$item->applyDamage(1);
+					$this->fire();
+				}
+				return true;
+			}
+			if($item instanceof Shovel && !$this->extinguished){
+				$item->applyDamage(1);
+				$this->extinguish();
+				return true;
+			}
+
+			if($this->addItem(clone $item)){
+				$item->pop();
+				$this->position->getWorld()->addSound($this->position->add(0.5, 0.5, 0.5), new ItemFrameAddItemSound());
+				$this->position->getWorld()->setBlock($this->position, $this);
+				if(count($this->items) === 1){
+					$this->position->getWorld()->scheduleDelayedBlockUpdate($this->position, 1);
+				}
+				return true;
+			}
+		}
+		return false;
+	}
+
+	public function onNearbyBlockChange() : void{
+		$block = $this->getSide(Facing::UP);
+		if($block instanceof Water && !$this->extinguished){
+			$this->extinguish();
+		}
+	}
+
+	public function onEntityInside(Entity $entity) : bool{
+		if($this->extinguished){
+			if($entity->isOnFire()){
+				$this->fire();
+				return true;
+			}
+			return false;
+		}
+		if($entity instanceof SplashPotion && $entity->getPotionType()->getDisplayName() === PotionType::WATER()->getDisplayName()){
+			$this->extinguish();
+			return true;
+		}elseif($entity instanceof Living){
+			$entity->attack(new EntityDamageByBlockEvent($this, $entity, EntityDamageEvent::CAUSE_FIRE, 1));
+			$entity->setOnFire(8);
+		}
+		return false;
+	}
+
+	public function onScheduledUpdate() : void{
+		if(!$this->extinguished){
+			foreach($this->items as $slot => $item){
+				$this->increaseSlotTime($slot);
+				if($this->getSlotTime($slot) >= $this->getFurnaceType()->getCookDurationTicks()){
+					$this->setItem(VanillaItems::AIR(), $slot);
+					$this->setSlotTime($slot, 0);
+					$result = ($item = $this->position->getWorld()->getServer()->getCraftingManager()->getFurnaceRecipeManager($this->getFurnaceType())->match($item)) instanceof FurnaceRecipe ? $item->getResult() : VanillaItems::AIR();
+					$this->position->getWorld()->dropItem($this->position->add(0, 1, 0), $result);
+				}
+			}
+			$this->position->getWorld()->setBlock($this->position, $this);
+			if(!empty($this->items)){
```

Comment:
Following phpstan's analysis, this needs to be a more strict comparison. Don't to use empty() when you can check the count of items